File: Lab1/Lab1b/Plane.cpp

#include <string>
#include "Plane.h"



Plane::Plane(string b, int q, float p)
{
	this->brand = b;
	this->quantity = q;
	this->price = p;
	this->result = 0;
}

string Plane::getBrand()
{
	return this->brand;
}

int Plane::getQuantity()
{
	return this->quantity;
}

float Plane::getPrice()
{
	return this->price;
}

int Plane::getResult()
{
	return this->result;
}

void Plane::addResult(int res)
{
	this->result = res;
}

int Plane::CountFibonacci(int quantity)
{
	if (quantity <= 1)
		return quantity;
	return CountFibonacci(quantity - 1) + CountFibonacci(quantity - 2);
}

Plane::Plane()
{
}

File: Lab1/Lab1a/Lab1a/Plane.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Lab1a
{
    class Plane
    {
        public string Name { get; set; }
        public int Quantity { get; set; }
        public double Price { get; set; }
        public int Result { get; set; }


        //public Plane(string name, int quantity, double price)
        //{
        //    this.Name = name;
        //    this.Quantity = quantity;
        //    this.Price = price;
        //}




        public int CountFibonacci(int number)
        {
            if (number == 0) return 0; //To return the first Fibonacci number   
            if (number == 1) return 1; //To return the second Fibonacci number   
            return CountFibonacci(number - 1) + CountFibonacci(number - 2);
        }

        internal int CompareTo(Plane item)
        {
            return this.Result.CompareTo(item.Result);
        }

        public int ReturnResult()
        {
            return CountFibonacci(this.Quantity);
        }
    }
}
